Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/interfaces.go

```diff
@@ -16,6 +16,24 @@ limitations under the License.
 
 package runtime
 
+// ObjectScheme represents common conversions between formal external API versions
+// and the internal Go structs. ObjectScheme is typically used with ObjectCodec to
+// transform internal Go structs into serialized versions. There may be many valid
+// ObjectCodecs for each ObjectScheme.
+type ObjectScheme interface {
+	ObjectConvertor
+	ObjectTyper
+	ObjectCreater
+	ObjectCopier
+}
+
+// ObjectCodec represents the common mechanisms for converting to and from a particular
+// binary representation of an object.
+type ObjectCodec interface {
+	ObjectEncoder
+	Decoder
```

Comment:
----- Original Message -----
> > +// ObjectScheme represents common conversions between formal external API
> > versions
> > +// and the internal Go structs. ObjectScheme is typically used with
> > ObjectCodec to
> > +// transform internal Go structs into serialized versions. There may be
> > many valid
> > +// ObjectCodecs for each ObjectScheme.
> > +type ObjectScheme interface {
> > +	ObjectConvertor
> > +	ObjectTyper
> > +	ObjectCreater
> > +	ObjectCopier
> > +}
> > +
> > +// ObjectCodec represents the common mechanisms for converting to and from
> > a particular
> > +// binary representation of an object.
> > +type ObjectCodec interface {
> > +	ObjectEncoder
> > +	Decoder
> 
> Why not an `ObjectDecoder`?  We're allowing encoding to various versions (the
> difference between `Encoder` and `ObjectEncoder`), why provide methods to
> inspecting encoded json and determining what it is?
> 

ObjectCodec is the minimal set of methods needed to create a Codec